src/Nancy.Serialization.JsonNet/JsonNetBodyDeserializer.cs

```diff
@@ -52,8 +52,16 @@ public object Deserialize(string contentType, Stream bodyStream, BindingContext
         {
             var deserializedObject = 
                 this.serializer.Deserialize(new StreamReader(bodyStream), context.DestinationType);
+
+            var properties = 
+                context.DestinationType.GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                    .Select(p => new BindingMemberInfo(p));
             
-            if (context.DestinationType.GetProperties(BindingFlags.Public | BindingFlags.Instance).Except(context.ValidModelProperties).Any())
+            var fields = 
+                context.DestinationType.GetFields(BindingFlags.Public | BindingFlags.Instance)
+                    .Select(f => new BindingMemberInfo(f));
+
+            if (properties.Concat(fields).Except(context.ValidModelBindingMembers).Any())

```

Comment:
Does this work? Has the `Equals` method been overridden on the `BindingMemberInfo` type?